filebeat/harvester/log.go

```diff
@@ -99,6 +102,19 @@ func (h *Harvester) Harvest() {
 			continue
 		}
 
+		if len(h.ExcludeLinesRegexp) > 0 {
+			if MatchRegExps(h.ExcludeLinesRegexp, text) {
+				// drop line
+				continue
```

Comment:
using continue here is righteous dangerous. When line is dropped, file offset is broken + backoff algorithm is not properly reset.

Maybe move generation of FileEvent and forwarding into it's own function. The check for generating/sending the event will be made in this extra function.